### PR TITLE
Adds Repo Template and Terraform Repo Template

### PR DESCRIPTION
**Pull request by ChrisScottThomas**
Creating template repositories to base new repos from.

### CODE REVIEW

File: repo-template.tf [hunk 1]
```diff
@@ -0,0 +1,34 @@
+resource "github_repository" "dataworks_repo_template" {
+  name        = "dataworks_repo_template"
```
Comment (dan-hill2802) [inline repo-template.tf:2]: Can the names be hyphens instead of underscores like all the others?

Comment (ChrisScottThomas) [reply repo-template.tf:2]: Done.

**Comment by ChrisScottThomas** [comment]
> Considering we might develop more templates over time, it would be better to name it `repo-template-terraform.tf`

Yep, makes sense.

File: repo-template-terraform.tf [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+}
+
+resource "github_branch_protection" "dataworks-terraform-repo-template_master" {
+  branch         = "${github_repository.dataworks-terraform-repo-template.default_branch}"
+  repository     = "${github_repository.dataworks-terraform-repo-template.name}"
+  enforce_admins = false
+
+  required_status_checks {
+    strict = true
+    # The contexts line should only be kept for Terraform repos.
```
Comment (dan-hill2802) [inline repo-template-terraform.tf:28]: I guess this comment is redundant for this file?

File: repo-template-terraform.tf [hunk 3]
```diff
@@ -0,0 +1,36 @@
+resource "github_repository" "dataworks-terraform-repo-template" {
+  name        = "dataworks-terraform-repo-template"
+  description = "Template Terraform repository for Dataworks GitHub"
```
Comment (dan-hill2802) [inline repo-template-terraform.tf:3]: It's DataWorks

File: repo-template.tf [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  description = "Template repository for Dataworks GitHub"
```
Comment (dan-hill2802) [inline repo-template.tf:3]: It's DataWorks

Comment (ChrisScottThomas) [reply repo-template.tf:3]: Done

File: repo-template-terraform.tf [hunk 5]
```diff
@@ -0,0 +1,35 @@
+resource "github_repository" "dataworks-terraform-repo-template" {
+  name        = "dataworks-terraform-repo-template"
+  description = "Template Terraform repository for DataWorks GitHub"
+  auto_init   = true
+  is_template = true
+
+  allow_merge_commit = false
+  has_issues         = true
```
Comment (mattburgess) [inline repo-template-terraform.tf:8]: Point of contention this, I think. I don't like us being able to manage issues in 2 places, so I think this should be false. I know @dan-hill2802 has a dissenting opinion though. I do like the idea of managing issues with the code, but that's not where we're at now, or likely to ever get to. JIRA, for better or worse, is how we manage issues.

File: repo-template-terraform.tf [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+resource "github_branch_protection" "dataworks-terraform-repo-template_master" {
+  branch         = "${github_repository.dataworks-terraform-repo-template.default_branch}"
+  repository     = "${github_repository.dataworks-terraform-repo-template.name}"
+  enforce_admins = false
+
+  required_status_checks {
+    strict   = true
+    contexts = ["concourse-ci/status"]
```
Comment (mattburgess) [inline repo-template-terraform.tf:28]: Nice, thanks!

File: repo-template.tf [hunk 7]
```diff
@@ -0,0 +1,34 @@
+resource "github_repository" "dataworks_repo_template" {
+  name        = "dataworks_repo_template"
+  description = "Template repository for DataWorks GitHub"
+  auto_init   = true
+  is_template = true
+
+  allow_merge_commit = false
+  has_issues         = true
```
Comment (mattburgess) [inline repo-template.tf:8]: Again, set this to `false` please.

File: repo-template-terraform.tf (hunk 5 above)
Comment (dan-hill2802) [reply repo-template-terraform.tf:8]: Issues aren't just for us, we should allow other consumers to report issues

**Pull request merged**